src/main/java/rx/swing/sources/ChangeEventSource.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright 2015 Netflix, Inc.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package rx.swing.sources;
+
+import rx.Observable;
+import rx.Observable.OnSubscribe;
+import rx.Subscriber;
+import rx.functions.Action0;
+import rx.schedulers.SwingScheduler;
+import rx.subscriptions.Subscriptions;
+
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+
+public enum ChangeEventSource { ; // no instances
+
+    /**
+     * Creates an observable corresponding to change events (e.g. progressbar value changes).
+     *
+     * Due to the lack of a common interface in Java (up to at least version 8), the implementation
+     * is generic and uses internally reflection to add and remove it's {@link ChangeListener}'s.
+     * The contract is therefor that the given parameter object MUST have the typical two public methods "addChangeListener"
+     * (like {@link javax.swing.JProgressBar#addChangeListener(ChangeListener)}) and "removeChangeListener"
+     * (like {@link javax.swing.JProgressBar#removeChangeListener(ChangeListener)}).
+     *
+     * For more info to change listeners and events see <a href="https://docs.oracle.com/javase/tutorial/uiswing/events/changelistener.html">
+     * How to Write a Change Listener</a>.
+     *
+     * @param changeEventSource
+     * 		The object to register the observable for.
+     * @return Observable emitting the change events.
+     * @throws IllegalArgumentException if the given parameter object has not the needed signature
+     */
+    public static Observable<ChangeEvent> fromChangeEventsOf(final Object changeEventSource) {
+        checkHasChangeListenerSupport(changeEventSource);
+        return Observable.create(new OnSubscribe<ChangeEvent>() {
+            @Override
+            public void call(final Subscriber<? super ChangeEvent> subscriber) {
+                final ChangeListener listener = new ChangeListener() {
+
```

Comment:
Since you're likely making some edits to this file, might as well remove this blank line and the one below.